apps/opik-frontend/src/components/pages/TracesPage/MetricsTab/MetricsTab.tsx

```diff
@@ -0,0 +1,203 @@
+import { Button } from "@/components/ui/button";
+import { ChartLine as ChartLineIcon } from "lucide-react";
+import React, { useEffect, useMemo, useRef, useState } from "react";
+import SelectBox from "@/components/shared/SelectBox/SelectBox";
+import MetricChart from "@/components/pages/TracesPage/MetricsTab/MetricChart/MetricChart";
+import useProjectMetric, {
+  IntervalType,
+} from "@/api/projects/useProjectMetric";
+import last from "lodash/last";
+import dayjs from "dayjs";
+import { StringParam, useQueryParam, withDefault } from "use-query-params";
+import RequestChartDialog from "@/components/pages/TracesPage/MetricsTab/RequestChartDialog/RequestChartDialog";
+import useTracesOrSpansList, {
+  TRACE_DATA_TYPE,
+} from "@/hooks/useTracesOrSpansList";
+import NoTracesPage from "@/components/pages/TracesPage/NoTracesPage";
+
+const DAYS_OPTIONS = [
+  {
+    value: "1",
+    label: "1 day",
+  },
+  {
+    value: "3",
+    label: "3 days",
+  },
+  {
+    value: "7",
+    label: "7 days",
+  },
+  {
+    value: "30",
+    label: "30 days",
+  },
+];
+
+const POSSIBLE_DAYS_OPTIONS = DAYS_OPTIONS.map((dayOption) => dayOption.value);
+
+const DEFAULT_DAYS_VALUE = last(DAYS_OPTIONS)!.value;
+
+interface MetricsTabProps {
+  projectId: string;
+}
+
+const nowUTC = dayjs().utc();
+const intervalEnd = nowUTC.format();
+
+const MetricsTab = ({ projectId }: MetricsTabProps) => {
+  const [days, setDays] = useQueryParam(
+    "days",
+    withDefault(StringParam, DEFAULT_DAYS_VALUE),
+  );
+  const [requestChartOpen, setRequestChartOpen] = useState(false);
+
+  // to show if there is no data
+  const { data: traces } = useTracesOrSpansList(
```

Comment:
Didn't you consider moving the data request functionality to the chart component? From my point of view, it would be cleaner.